app/kuma-dp/pkg/dataplane/accesslogs/server.go

```diff
@@ -4,45 +4,37 @@ import (
 	"fmt"
 	"io"
 	"net"
-	"strconv"
-	"strings"
 	"sync/atomic"
-	"time"
 
-	"github.com/go-logr/logr"
-	"github.com/golang/protobuf/ptypes"
-	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 
 	kumadp "github.com/Kong/kuma/pkg/config/app/kuma-dp"
 	"github.com/Kong/kuma/pkg/core"
-	util_proto "github.com/Kong/kuma/pkg/util/proto"
+	"github.com/pkg/errors"
 
-	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
-	envoy_data_accesslog_v2 "github.com/envoyproxy/go-control-plane/envoy/data/accesslog/v2"
-	v2 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	envoy_accesslog "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
 )
 
 var logger = core.Log.WithName("accesslogs-server")
 
-const (
-	defaultConnectTimeout = 5 * time.Second
-)
-
 type accessLogServer struct {
-	server *grpc.Server
+	server     *grpc.Server
+	newHandler logHandlerFactoryFunc
+	newSender  logSenderFactoryFunc
 
 	// streamCount for counting streams
 	streamCount int64
 }
 
 func NewAccessLogServer() *accessLogServer {
 	return &accessLogServer{
-		server: grpc.NewServer(),
+		server:     grpc.NewServer(),
+		newHandler: defaultHandler,
+		newSender:  defaultSender,
```

Comment:
I think server should receive prepared handler (handler factory) and not care about sender. Handler could just as well log to file, log to stdout or do whatever.

Take a look at this:
[patch.txt](https://github.com/Kong/kuma/files/4244329/patch.txt)

Reply:
I applied the patch